### PR TITLE
New map merging tool.

### PR DESCRIPTION
**Pull request by xxalpha**
Map merger written in python. Supports tgm format (see https://github.com/tgstation/-tg-station/pull/10623).

Kept old map merger just in case.

### CODE REVIEW

**Comment by optimumtact** [comment]
:+1:

File: tools/mapmerge/dmm2tgm.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                                        if in_quote_block:
+                                                in_quote_block = False
+                                        else:
+                                                in_quote_block = True
+
+                        converted_map = converted_map + char
+
+        #OVERWRITE MAP FILE WITH CONVERTED MAP STRING
+        with open(map_file, "r+") as final_converted_map:
+                final_converted_map.write("//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE \n")
```
Comment (neersighted) [inline tools/mapmerge/dmm2tgm.py]: Put the magic header in a var please

File: tools/mapmerge/mapmerger.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+        print(list_of_files[i])
+    merge = input("\nPress Enter to merge...")
+    if merge == "abort":
+        print("\nAborted map merge.")
+        sys.exit()
+    else:
+        for i in valid_indices:
+            try:
+                print("MERGING: {}".format(str(list_of_files[i])))
+                if merge_map(str(list_of_files[i]), str(list_of_files[i]) + ".backup") != 1:
```
Comment (neersighted) [inline tools/mapmerge/mapmerger.py]: Cache the `str()` representation in a var

**Comment by Iamgoofball** [comment]
merge when @Cheridan @ChangelingRain @WJohn @Aranclanos @KorPhaeron @optimumtact @d3athrow @N3X15

**Comment by xxalpha** [comment]
You didn't ping remie and why are you pinging /vg/?

**Comment by Iamgoofball** [comment]
@RemieRichards @xxalpha @xxalpha @xxalpha @Fox-McCloud

**Comment by optimumtact** [comment]
For maximum meme

**Comment by duncathan** [comment]
Is this good to merge? I don't know python

**Comment by optimumtact** [comment]
It's not the cleanest python I've ever seen, but it works

**Comment by xxalpha** [comment]
It's the cleanest python I've ever seen. (it's the only python I've ever seen)

**Comment by Steelpoint** [comment]
Am I a idiot doing something wrong because mapmerger ain't working for me anymore.

Whenever I execute the "Run Map Merge" program it just comes up with 'Python' is not recognised as an internal or external command, operable program or batch file

**Comment by optimumtact** [comment]
you should install python

**Comment by optimumtact** [comment]
https://www.python.org/downloads/release/python-351/

**Comment by optimumtact** [comment]
x86-64 installer is what you will want for windows

**Comment by Steelpoint** [comment]
Oh for fucks sake what was wrong with the old merger??

Whatever.

**Comment by optimumtact** [comment]
it was uncommented and badly understood it also did not work with the tgm format that remie brainstormed up to reduce merge conflicts in the map files.

**Comment by xxalpha** [comment]
It's literally the same usage. Prepare maps.bat -> edit -> Map merge.bat

It's even the same names.

**Comment by Steelpoint** [comment]
Even with python it won't work.

Its not that I don't understand what to do, its that the program is not working even with python installed.

**Comment by optimumtact** [comment]
sounds like python isn't on your path, which is strange, because the 3.5 install should add it automatically.

https://docs.python.org/3/using/windows.html

Did you tick add python to the path option when using the installer?

**Comment by optimumtact** [comment]
in fact install now should do it automatically

**Comment by xxalpha** [comment]
Steelpoint when installing python, check the "add python to PATH" box.

You can reinstall and do that.

**Comment by optimumtact** [comment]
@xxalpha isn't the windows python launcher called py ?

Might be the cause

**Comment by Steelpoint** [comment]
You guys really should have left actual documentation in the map merge folder on how to set this up.

What's the point in some fancy rewrite if someone new coming along does not even understand what's going on.

**Comment by xxalpha** [comment]
It's a beta.

**Comment by Steelpoint** [comment]
Its still not working. I got python to work and the mapmerger runs and all but the number of changes to the files I've edited do not lower when I run mapmerger.

**Comment by RemieRichards** [comment]
@xxalpha py2exe it. then it can run on any version of Windows, without a python install.
(its what I did for dmm2tgm)

**Comment by optimumtact** [comment]
@Steelpoint all files will show an extremely large diff at first because they are converted to the tgm format

that's my understanding at least.

**Comment by Steelpoint** [comment]
So what's the point in a mapmerger that does not condense the lines changed?

**Comment by RemieRichards** [comment]
@Steelpoint it does, this is just the first run conversion to the new format which makes map conflicts less likely, I assume you would like less map conflicts right?

**Comment by Steelpoint** [comment]
So no one care's that the map's going to change 60000 lines now?

**Comment by xxalpha** [comment]
The first run of this is going to be big because it's converting to tgm format by default, which will reduce conflict occurrence in future edits.

**Comment by optimumtact** [comment]
THINK ABOUT YOU DONK LORD

**Comment by Steelpoint** [comment]
me no english

**Pull request merged**